kotlin-native/backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/objcexport/ObjCExport.kt

```diff
@@ -77,17 +79,28 @@ internal fun produceObjCExportInterface(
 }
 
 private class ObjCExportCompilerProblemCollector(val context: PhaseContext) : ObjCExportProblemCollector {
+    private val DeclarationDescriptor.psiLocation
+        get() = (this@psiLocation as? DeclarationDescriptorWithSource)?.source?.getPsi()?.let { MessageUtil.psiElementToMessageLocation(it) }
+
     override fun reportWarning(text: String) {
         context.reportCompilationWarning(text)
     }
 
     override fun reportWarning(declaration: DeclarationDescriptor, text: String) {
-        val psi = (declaration as? DeclarationDescriptorWithSource)?.source?.getPsi()
-                ?: return reportWarning(
-                        "$text\n    (at ${DescriptorRenderer.COMPACT_WITH_SHORT_TYPES.render(declaration)})"
-                )
+        val location = declaration.psiLocation ?: return reportWarning(
+                "$text\n    (at ${DescriptorRenderer.COMPACT_WITH_SHORT_TYPES.render(declaration)})"
+        )
 
-        val location = MessageUtil.psiElementToMessageLocation(psi)
+        context.messageCollector.report(CompilerMessageSeverity.WARNING, text, location)
+    }
+
+    override fun reportError(text: String) {
+        context.reportCompilationError(text)
```

Comment:
Here is the problem: this function reports the error and throws an exception. Meaning, it hides all other errors that would be reported later otherwise.